internal/docker/docker.go

```diff
@@ -0,0 +1,190 @@
+package docker
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+	"time"
+
+	containerTypes "github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/api/types/image"
+	"github.com/docker/docker/api/types/mount"
+	"github.com/docker/docker/client"
+	"github.com/konstructio/colony/internal/constants"
+	"github.com/konstructio/colony/internal/logger"
+)
+
+type Client struct {
+	cli *client.Client
+	log *logger.Logger
+}
+
+func New(logger *logger.Logger) (*Client, error) {
+	cli, err := client.NewClientWithOpts(client.FromEnv, client.WithAPIVersionNegotiation())
+	if err != nil {
+		return nil, fmt.Errorf("error creating docker client: %w ", err)
+	}
+
+	return &Client{
+		cli: cli,
+		log: logger,
+	}, nil
+}
+
+// func getColonyK3sContainerIDAndName(ctx context.Context, c *Client) {
+func getColonyK3sContainerIDAndName(ctx context.Context, c *Client) (string, string, error) {
+
+	containers, err := c.cli.ContainerList(ctx, containerTypes.ListOptions{All: true})
+	if err != nil {
+		return "", "", fmt.Errorf("error listing containers on host: %w ", err)
+	}
+
+	for _, container := range containers {
+		if container.Names[0] == "/"+constants.ColonyK3sContainerName {
+			return container.ID, container.Names[0], nil
+		}
+	}
+	return "", "", fmt.Errorf("container %q not found", constants.ColonyK3sContainerName)
+}
+
+func (c *Client) RemoveColonyK3sContainer(ctx context.Context) error {
+
+	defer c.cli.Close()
+
+	colonyK3sContainerID, colonyK3sContainerName, err := getColonyK3sContainerIDAndName(ctx, c)
+	if err != nil {
+		return fmt.Errorf("error getting %q container: %w ", constants.ColonyK3sContainerName, err)
+	}
+	c.log.Info(fmt.Sprintf("found container name %q with ID %q  ", strings.TrimPrefix(colonyK3sContainerName, "/"), colonyK3sContainerID[:constants.DefaultDockerIDLength]))
```

Comment:
iirc the logger has a "printf" style formatter

```suggestion
	c.log.Infof("found container name %q with ID %q  ", strings.TrimPrefix(colonyK3sContainerName, "/"), colonyK3sContainerID[:constants.DefaultDockerIDLength])
```